src/cockpit/packages.py

```diff
@@ -423,25 +459,9 @@ def get_condition_files(cls) -> Iterable[str]:
         for manifest in cls.load_manifests():
             yield from manifest.get_condition_files()
 
-    def check_condition(self, condition: str, value: object) -> bool:
-        check_fn = self.CONDITIONS[condition]
-
-        # All known predicates currently only work on strings
-        if not isinstance(value, str):
-            return False
-
-        return check_fn(value)
-
     def check_conditions(self, manifest: Manifest) -> bool:
         for condition in manifest.conditions:
-            try:
-                okay = self.check_condition(condition.name, condition.value)
-            except KeyError:
-                # do *not* ignore manifests with unknown predicates, for forward compatibility
-                logger.warning('  %s: ignoring unknown predicate in manifest: %s', manifest.path, condition.name)
-                continue
-
-            if not okay:
+            if not condition.check(self.path_exists):
                 logger.debug('  hiding package %s as its %s condition is not met', manifest.path, condition)
```

Comment:
This message is still displaying raw conditions, unfortunately.  But it's also a debug.  I think we can let that one slide...

Reply:
Fixed. Not super pretty, but good enough for a debug message IMHO:

> hiding package /test as its DisjunctiveCondition(PathExistsCondition('/missing', exists=True), PathExistsCondition('/existing', exists=False)) condition is not met